modules/fbs-core/web/src/app/dialogs/new-checker-dialog/new-checker-dialog.component.ts

```diff
@@ -58,10 +62,46 @@ export class NewCheckerDialogComponent implements OnInit {
       );
       this.checkerForm.controls["ord"].setValue(this.checker.ord);
     }
-    this.courseId = this.data.courseId;
-    this.taskId = this.data.taskId;
-    this.setDefaultValues();
-    this.choosedSQLChecker = false;
+
+    if (this.checker.mainFileUploaded == true) {
+      this.checkerService
+        .getChecker(this.courseId, this.taskId)
+        .pipe(map((checkers) => checkers[0]))
```

Comment:
Here the main file is loaded only for the first checker. With several checkers, the content of the main file of the first checker is always returned. Although you can't download this at the moment, I would still adapt it.

Reply:
generic optioan applied